frontend/components/forms/LoginForm/LoginForm.tsx

```diff
@@ -40,6 +46,11 @@ const LoginForm = ({
     email: "",
     password: "",
   });
+  const [showPendingEmail, setShowPendingEmail] = useState(pendingEmail);
+
+  useEffect(() => {
+    setShowPendingEmail(pendingEmail);
+  }, [pendingEmail]);
```

Comment:
Is this `useEffect` necessary? If `pendingEmail` updates, won't the component rerender and
`showPendingEmail` be set to its initial state, `pendingEmail`, by nature of
`pendingEmail` being a prop?

Reply:
Even though the `pendingEmail` prop is updating to true, the `showPendingEmail` state isn't automatically updated to reflect this change. This is because state doesn't automatically update when props change. Without this useEffect, there's a  bug where we try to log in, it turns into a spinner, and then doesn't actually show the pending email component.

Reply:
I can leave a comment.